modules/snooker/src/main/ts/ephox/snooker/resize/Recalculations.ts

```diff
@@ -44,9 +44,23 @@ const recalculateWidth = function (warehouse: Warehouse, widths: number[]): Cell
   });
 };
 
-const recalculateHeight = function (warehouse: Warehouse, heights: number[]): CellHeightSpan[] {
+const recalculateWidthForColumns = (warehouse: Warehouse, widths: number[]): CellWidthSpan[] => {
+  const groups = Warehouse.justColumns(warehouse);
+
+  return Arr.map(groups, (col: SugarElement, index: number) => {
+    const colspan = col.dom.getAttribute('colspan');
```

Comment:
As mentioned in a previous comment, this would likely be better being a struct that already contained this information. It'll keep the code more consistent with the above.